WIP - [ShadowLayer] Fix initWithLayer: copying by adding sublayers.

stefanamiller opened the pull request:
Fixes the copy initWithLayer: initializer by adding the copied top and bottom shadow layers as sublayers.

The sublayers would never be added and a shadow would never appear. Also sets the delegate == self as per similar code in commonMDCShadowLayerInit.

stefanamiller committed:
[ShadowLayer] Fix copy initWithLayer: initializer by adding the copied top and bottom shadow layers as sublayers.

The sublayers would never be added and a shadow would never appear. Also sets the delegate == self as per similar code in commonMDCShadowLayerInit.

ianegordon commented:
Hi @stefanamiller , LGTM!

Do you have a before & after screenshot?

ianegordon reviewed (commented):
Waiting for CI before adding approval.
(I don't expect any issues)

stefanamiller committed:
[ShadowLayer] Adds testInitWithLayerCopy test.

Adds -testInitWithLayerCopy test for copying an MDCShadowLayer using -initWithLayer:. This adds a check to ensure the count of the sublayers is the same between the copy and the original. Also adds an assert in -testDefaultValues that the sublayers > 0.

stefanamiller commented:
Here are the comparison screenshots from the [testshadowlayercopy](https://github.com/material-components/material-components-ios/blob/testshadowlayercopy/components/ShadowLayer/examples/ShadowCopyController.swift) demo:

Before this change:
![Simulator Screen Shot - iPhone XS - 2019-03-14 at 14 32 17](https://user-images.githubusercontent.com/1054964/54394180-60bcb100-4669-11e9-8815-0246abc7b965.png)

After this change (55765ca):
![Simulator Screen Shot - iPhone XS - 2019-03-14 at 14 34 20](https://user-images.githubusercontent.com/1054964/54394181-60bcb100-4669-11e9-8106-a54d134a17ec.png)

ianegordon commented:
Marked as WIP as @stefanamiller and I are still discussing the implementation.

jverkoey commented:
Closing due to inactivity and an open offline discussion. Please respond here @stefanamiller / @ianegordon if you're still pursuing this change.
